src/lidar_visualizer/datasets/helipr.py

```diff
@@ -0,0 +1,132 @@
+                    for fn in os.listdir(self.scans_dir)
+                    if any(fn.endswith(ext) for ext in supported_file_extensions())
+                ]
+            ),
+            dtype=str,
+        )
+        if len(self.scan_files) == 0:
+            raise ValueError(f"Tried to read point cloud files in {self.scans_dir} but none found")
+        self.file_extension = self.scan_files[0].split(".")[-1]
+        if self.file_extension not in supported_file_extensions():
+            raise ValueError(f"Supported formats are: {supported_file_extensions()}")
+
+        # Obtain the pointcloud reader for the given data folder
+        if self.sequence_id.find("avia"):
+            self.dtype = np.dtype(
+                [
+                    ("x", np.float32),
+                    ("y", np.float32),
+                    ("z", np.float32),
+                    ("reflectivity", np.uint8),
+                    ("tag", np.uint8),
+                    ("line", np.uint8),
+                    ("offset_time", np.uint32),
+                ]
+            )
+        elif self.sequence_id.find("aeva"):
+            self.dtype = np.dtype(
+                [
+                    ("x", np.float32),
+                    ("y", np.float32),
+                    ("z", np.float32),
+                    ("reflectivity", np.float32),
+                    ("velocity", np.float32),
+                    ("time_offset_ns", np.int32),
+                    ("line_index", np.uint8),
+                    ("intensity", np.float32),
+                ]
+            )
+        elif self.sequence_id.find("ouster"):
+            self.dtype = np.dtype(
+                [
+                    ("x", np.float32),
+                    ("y", np.float32),
+                    ("z", np.float32),
+                    ("intensity", np.float32),
+                    ("t", np.int32),
+                    ("reflectivity", np.float16),
+                    ("ring", np.float16),
+                    ("ambient", np.float16),
+                ]
+            )
+        elif self.sequence_id.find("velodyne"):
+            self.dtype = np.dtype(
+                [
+                    ("x", np.float32),
+                    ("y", np.float32),
+                    ("z", np.float32),
+                    ("intensity", np.float32),
+                    ("ring", np.float16),
+                    ("time", np.float32),
+                ]
+            )
+        else:
+            print("[ERROR] Unsupported LiDAR Type")
+            sys.exit()
+
+    def __len__(self):
+        return len(self.scan_files)
+
+    def __getitem__(self, idx):
+        return self.read_point_cloud(self.scan_files[idx])
+
+    def read_point_cloud(self, file_path: str):
+        points = np.stack(
+            [
+                [line[0], line[1], line[2]]
+                for line in np.fromfile(file_path, dtype=self.dtype).tolist()
+            ]
+        )
+        scan = self.o3d.geometry.PointCloud()
```

Comment:
No intensity channel :cry: ?

Reply:
Yes please if you can add it :) you can check how I ~hacked~ did it in the generic one

Reply:
Done, let me know if this works!!

Reply:
you let me know :rofl: 
Will test later

Reply:
I tested it already, it works